src/emit/binary.rs

```diff
@@ -213,7 +213,7 @@ impl<'a> Binary<'a> {
         let (dibuilder, compile_unit) = module.create_debug_info_builder(
             true,
             inkwell::debug_info::DWARFSourceLanguage::C,
-            filename,
+            name,
```

Comment:
This argument to `create_debug_info_builder()` is the filename, changing it to the contract is not really the correct solution.

Having said that, this does hint towards the problem. Somehow the filename is not consistent.

Maybe on this line https://github.com/hyperledger/solang/blob/main/src/emit/mod.rs#L427 your new function `.filename()` should be used:


```rust
let filename = ns.files[self.loc.file_no()].filename();
```
That's a good idea anyway because of deterministic builds.

Reply:
Note that `self.loc` on this line should be the location of the contract, so if that is not producing the right file name, then there is an issue in location tracking.

Reply:
I tried this locally and I don't know why it works. In that case, the parent contract filename would be the filename of the child contract. (the child.ll file would change if I compile the contract from the CL, from when it is re-compiled when compiling the parent contract).
So my theory I mentioned on Discord is wrong :), could you explain this please? Thanks!

Reply:
Actually I think the problem is elsewhere. Both `create_contract.sol` and `runtime_errors.sol` have a contract definition for `child`.

Reply:
> Actually I think the problem is elsewhere. Both `create_contract.sol` and `runtime_errors.sol` have a contract definition for `child`.

But the definition for child is in the same file, so that will not cause a problem

Reply:
```
$ grep 'contract child' *.sol
create_contract.sol:contract child {
runtime_errors.sol:contract child {
```

Reply:
I think the problem lies in that the parent filename is the one being provided to all child contracts. What I did wrong was provide the contract name.
The right solution here would be providing the filename in which the contract is defined, I did that in  my latest commit. 
Thanks :)